src/components/App/App.js

```diff
@@ -1,18 +1,50 @@
 import './App.css';
 import banner from '../../images/old-school-theater-banner.png';
-import movieData from '../../data/data';
 import MoviesBox from '../MoviesBox/MoviesBox';
 import SingleMovie from '../SingleMovie/SingleMovie';
 import ExitMovie from '../ExitMovie/ExitMovie';
-import { useState } from 'react';
+import { useState, useEffect } from 'react';
+import { getAllMovies, getSingleMovie } from '../../apiCalls';
 
 function App() {
-  const [allMovies, setAllMovies] = useState(movieData.movies);
+  const [allMovies, setAllMovies] = useState([]);
   const [singleMovie, setSingleMovie] = useState(null);
+  const [movieNeeded, setMovieNeeded] = useState(null);
+  const [error, setError] = useState('')
+  const [loading, setLoading] = useState(false)
+
+  const fetchApi = (request, setter, id) => {
+    setLoading(true)
+    const apiCall = async () => {
+      try {
+        setter(await request(id))
+        setLoading(false)
+      } catch (error) {
+        setError(error)
+      }
+    }
+  
+    apiCall()   
+  }
+
+  useEffect(() => {
+    fetchApi(getAllMovies, setAllMovies)
+
+    return () => setError('')
+  }, [])
+
+  useEffect(() => {
+    if(movieNeeded) fetchApi(getSingleMovie, setSingleMovie, movieNeeded.id)
+
+    return () => {
+      setMovieNeeded(null)
+      setError('')
+    }
+  }, [movieNeeded])
 
   const viewMovie = (id) => {
     const selectedMovie = allMovies.find(movie => movie.id === id);
-    setSingleMovie(selectedMovie);
```

Comment:
I understand that we're setting state to the movie that has been found using the find method. I do not think we should do this. 
On line 37, we're then setting again the state for singleMovie from the API. So the state on this line will be rewritten from what we fetch, which also contains more information from the movie. If we have the dependency array for the `useEffect` on line 36 set to run each time the state of `movieNeeded` I think it would work.

Reply:
Actually, we might not even need to use find. We can probably pass the id directly into `movieNeeded` so that we can access it in the fetch method

Reply:
So, the ` setSingleMovie(selectedMovie);` was from the iteration with dummy data and that line has been removed and replaced with `setMovieNeeded(selectedMovie)` I am setting movie needed to the state that changes and is what our use effect runs off of. Let me know if you are thinking of something different but I think I am already doing what you are suggesting. 

I will remove the find and use just the ID for movieNeeded though, I think that is a good idea instead of carrying around the whole object!

Reply:
No, you understood correctly. The red didn't click as deleted when I was looking at this so I made the suggestion.